torch/lib/TensorLib/templates/Tensor.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+#include "TensorLib/Scalar.h"
+#include "TensorLib/Type.h"
+#include "TensorLib/TensorImpl.h"
+
+namespace tlib {
+class Type;
+class Tensor;
+
+// This allows non-const l-value and r-value references to be passed to tensor arguments:
+// void foo(TensorRef t) {} <-- pass a tensor by reference.
+// because this is a pass by reference it avoids ref-counting costs for arguments
+// which we know will not be destroyed.
```

Comment:
Sure, but you can also do that with a Tensor* or a Tensor&, we can make it clear in docs that TensorRef is not something that should be seen except in argument signatures.